lib/rouge/lexers/jsp.rb

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+    module Lexers
+        class JSP < HTML
+            desc 'JSP'
+            tag 'jsp'
+            filenames '*.jsp'
+            mimetypes 'text/x-jsp', 'application/x-jsp'
+
+            def initialize(*)
+                super
+                @java = Java.new
+            end
+
+            directives = %w(page include taglib)
+            actions = %w(scriptlet declaration expression)
+
+            prepend :script_content do
+                mixin :all_jsp_rules
+            end
+
+            prepend :style_content do
+                mixin :all_jsp_rules
+            end
+
+            prepend :root do
+                mixin :all_jsp_rules
+            end
+
+            prepend :dq do
+                mixin :jsp_interp_start
+                mixin :all_jsp_rules
+                rule /([^"]*?)(?=<%)/, Str
+                rule /([^"]*?)(?=<[a-zA-Z]*:[a-zA-Z]*)/, Str
+            end
+
+            prepend :tag do
+                mixin :all_jsp_rules
+            end
+       
+            state :all_jsp_rules do
+
+                rule(/<%--/) { token Comment; push :jsp_comment }
+
+                rule(/<%@\s*(#{directives.join('|')})\s*/) { token Name::Tag; push :jsp_directive }
+
+                rule(/<jsp:directive\.(#{directives.join('|')})/) { token Name::Tag; push :jsp_directive2 }
+
+                rule(/<jsp:(#{actions.join('|')})>/) { token Name::Tag; push :jsp_expression}
+
+                rule(/<[a-zA-Z]*:[a-zA-Z]*/) { token Name::Tag; push :jsp_tag_start }
+
+                rule(/<\/[a-zA-Z]*:[a-zA-Z]*/) { token Name::Tag; push :jsp_tag_end }
+
+                rule(/<%[!=]?/) { token Name::Tag; push :jsp_expression2 }
+            end
+
+            state :jsp_comment do
+                rule /(--%>)/, Comment, :pop!
+                rule /./m, Comment
+            end
+
+            state :jsp_directive do
+                rule /(%>)/, Name::Tag, :pop!
+        
+                mixin :tag
+            end
+
+            state :jsp_directive2 do
+                rule /(\/>)/, Name::Tag, :pop!
+
+                mixin :tag
+            end
+
+            state :jsp_expression do
+                rule(/<\/jsp:(#{actions.join('|')})>/) { token Name::Tag; pop! }
+                rule(/[^<\/]+/) { delegate @java }

```

Comment:
Although there's technically nothing wrong with this syntax, let's stick to `do ... end` style blocks. I don't see this style used anywhere else in the code so let's stay consistent.

Reply:
??? i use this style all the time...

Reply:
Ah, yep. I see it now. Don't know why I wasn't finding these examples earlier.